nodejs/README.md

```diff
@@ -29,25 +29,44 @@ The following Node.js code snippet demonstrates downloading a GTFS-realtime
 data feed from a particular URL, parsing it as a FeedMessage (the root type of
 the GTFS-realtime schema), and iterating over the results.
 
+In order to make this example work, you must first install `node-fetch` with NPM.
+
+_Note: this exemple is using ES modules (`import`/`export` syntax) and is not compatible
+with CommonJS (`require` syntax). You can use CommonJS by converting `import` to `require`
+and installing `node-fetch@2`. Learn more about ES modules [here](https://nodejs.org/api/esm.html)._
+
 ```javascript
-var GtfsRealtimeBindings = require('gtfs-realtime-bindings');
-var request = require('request');
-
-var requestSettings = {
-  method: 'GET',
-  url: 'URL OF YOUR GTFS-REALTIME SOURCE GOES HERE',
-  encoding: null
-};
-request(requestSettings, function (error, response, body) {
-  if (!error && response.statusCode == 200) {
-    var feed = GtfsRealtimeBindings.transit_realtime.FeedMessage.decode(body);
-    feed.entity.forEach(function(entity) {
+import GtfsRealtimeBindings from "gtfs-realtime-bindings";
+import fetch from "node-fetch";
+
+fetch("<GTFS-realtime source URL>", {
+  headers: {
+    "x-api-key": "<redacted>",
+    // replace with your GTFS-realtime source's auth token
+    // e.g. x-api-key is the header value used for NY's MTA GTFS APIs
+  },
+})
+  .then((response) => {
+    if (!res.ok) {
+      console.log("Error: " + res.status);
```

Comment:
```suggestion
      const error = new Error(`${res.url}: ${res.status} ${res.statusText}`);
      error.response = res;
      throw error;
```

Reply:
I didnt used async to avoid top-level async in the example (even if this is now supported with ESM).
I think it could still cause problems if someone use this with CJS as described in the example note.

Reply:
(I assume you didn't mean to comment here? Because my suggestion to `throw` a proper `Error` still applies to `.then()`-style code.)

Even if we don't use top-level `await`, we would use an async immediately invoked function expression (IIFEE) as an "intermediate step" towards top-level `await`:

```js
;(async () => {
	// …
})()
.catch((error) => {
	console.error(error);
	process.exit(1);
})
```